### PR TITLE
feat: sort wallets [OTE-825]

### PR DESCRIPTION
**Pull request by yogurtandjam**
sorts wallets according to spec https://linear.app/dydx/issue/OTE-825/sort-for-wallets-show-default-wallets-and

- metamask
- phantom
- keplr
^ these three will go first and default to a download link if the extension isn't already installed

- walletconnect
- coinbase
- email or social (privy)
- coinbase
- okx wallet
  - if the okx wallet is not installed, we'll still show the logo and just route it through the walletconnect option. if it is installed, we'll display as part of injected wallets. this isn't exactly how it was outlined in the spec so will want to get confirmation that this is acceptable behavior

image

https://www.loom.com/share/01711f85a43045fc9e252f91ccf3b352

### CODE REVIEW

File: src/hooks/useDisplayedWallets.ts [hunk 1]
```diff
@@ -51,45 +60,60 @@ export const useDisplayedWallets = (): WalletInfo[] => {
...
+          icon: injectedMetaMask.detail.info.icon,
+          name: injectedMetaMask.detail.info.name,
+          rdns: injectedMetaMask.detail.info.rdns,
+        }
+      : {
+          connectorType: ConnectorType.DownloadWallet,
+          name: WalletType.MetaMask,
+          downloadLink: METAMASK_DOWNLOAD_LINK,
+        };
+
```
Comment (tinaszheng) [inline src/hooks/useDisplayedWallets.ts:75]:
```suggestion
    const metamaskWallet = injectedMetaMask ?? {
          connectorType: ConnectorType.DownloadWallet,
          name: WalletType.MetaMask,
          downloadLink: METAMASK_DOWNLOAD_LINK,
        };

```

File: src/hooks/useDisplayedWallets.ts [hunk 2]
```diff
@@ -24,15 +27,21 @@ export const useDisplayedWallets = (): WalletInfo[] => {
...
       injectedWallets.findIndex((wallet) => wallet.detail.info.rdns === OKX_MIPD_RDNS) !== -1;
 
+    const injectedMetaMask = injectedWallets.find(
+      (wallet) => wallet.detail.info.rdns === METAMASK_MIPD_RDNS
+    );
+
     const enabledInjectedWallets = injectedWallets
       .filter(
         (wallet) =>
+          // Remove Metamask. We will show it no matter what at the front of
```
Comment (tinaszheng) [inline src/hooks/useDisplayedWallets.ts:43]: fix grammar? 😆

Comment (yogurtandjam) [reply src/hooks/useDisplayedWallets.ts:43]: lmao, this comment got mangled 😆 what happened to the second line?!

File: src/hooks/useDisplayedWallets.ts (hunk 1 above)
Comment (yogurtandjam) [reply src/hooks/useDisplayedWallets.ts]: `injectedMetaMask` isn't the right format D: it needs to go through the same transformation that we have in the `.map` function

**Review by tinaszheng — APPROVED** [review]

**Pull request merged**